In [27]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

In [28]:
df = pd.read_csv('marketing_df.csv')

In [29]:
df.head()

Unnamed: 0  Unnamed: 0.1  Education  Marital_Status   Income  Kidhome  \
0           0             0          2               2  84835.0        0   
1           1             1          2               4  57091.0        0   
2           2             2          2               3  67267.0        0   
3           3             3          2               5  32474.0        1   
4           4             4          2               4  21474.0        1   

   Teenhome  Recency  MntWines  MntFruits  ...  AcceptedCmp1  AcceptedCmp2  \
0         0        0       189        104  ...             0             0   
1         0        0       464          5  ...             0             1   
2         1        0       134         11  ...             0             0   
3         1        0        10          0  ...             0             0   
4         0        0         6         16  ...             0             0   

   Response  Complain  Country  Children  Age  Total_Spent  customer_lifetime  \
0         1         0        6         0   51         1190                6.7   
1         1         0        1         0   60          577                6.7   
2         0         0        7         1   63          251                6.7   
3         0         0        0         2   54           11                6.8   
4         1         0        6         1   32           91                6.8   

   cluster  
0        0  
1        1  
2        0  
3        3  
4        3  

[5 rows x 32 columns]

In [30]:
df.drop(['Children'],axis=1, inplace=True)

In [31]:
df.head()

Unnamed: 0  Unnamed: 0.1  Education  Marital_Status   Income  Kidhome  \
0           0             0          2               2  84835.0        0   
1           1             1          2               4  57091.0        0   
2           2             2          2               3  67267.0        0   
3           3             3          2               5  32474.0        1   
4           4             4          2               4  21474.0        1   

   Teenhome  Recency  MntWines  MntFruits  ...  AcceptedCmp5  AcceptedCmp1  \
0         0        0       189        104  ...             0             0   
1         0        0       464          5  ...             0             0   
2         1        0       134         11  ...             0             0   
3         1        0        10          0  ...             0             0   
4         0        0         6         16  ...             0             0   

   AcceptedCmp2  Response  Complain  Country  Age  Total_Spent  \
0             0         1         0        6   51         1190   
1             1         1         0        1   60          577   
2             0         0         0        7   63          251   
3             0         0         0        0   54           11   
4             0         1         0        6   32           91   

   customer_lifetime  cluster  
0                6.7        0  
1                6.7        1  
2                6.7        0  
3                6.8        3  
4                6.8        3  

[5 rows x 31 columns]

In [33]:
from sklearn.preprocessing import StandardScaler

In [39]:
marketing_df = df.drop(columns=['Response'])
transformer = StandardScaler().fit(marketing_df)
marketing_df_standardized = transformer.transform(marketing_df)
X = pd.DataFrame(marketing_df_standardized, columns = marketing_df.columns)

print(X.head())

   Unnamed: 0  Unnamed: 0.1  Education  Marital_Status    Income   Kidhome  \
0   -1.731268     -1.729064  -0.352454       -1.602398  1.294980 -0.823208   
1   -1.729703     -1.727516  -0.352454        0.254202  0.192843 -0.823208   
2   -1.728138     -1.725969  -0.352454       -0.674098  0.597087 -0.823208   
3   -1.726572     -1.724421  -0.352454        1.182503 -0.785073  1.039532   
4   -1.725007     -1.722874  -0.352454        0.254202 -1.222051  1.039532   

   Teenhome   Recency  MntWines  MntFruits  ...  AcceptedCmp4  AcceptedCmp5  \
0 -0.929307 -1.693697 -0.344435   1.955247  ...     -0.282912     -0.280107   
1 -0.929307 -1.693697  0.471034  -0.536764  ...     -0.282912     -0.280107   
2  0.908545 -1.693697 -0.507529  -0.385733  ...     -0.282912     -0.280107   
3  0.908545 -1.693697 -0.875231  -0.662623  ...     -0.282912     -0.280107   
4 -0.929307 -1.693697 -0.887093  -0.259874  ...     -0.282912     -0.280107   

   AcceptedCmp1  AcceptedCmp2  Complain   Country       

In [40]:
X.drop(['cluster'],axis=1, inplace=True)

In [41]:
X.head()

Unnamed: 0  Unnamed: 0.1  Education  Marital_Status    Income   Kidhome  \
0   -1.731268     -1.729064  -0.352454       -1.602398  1.294980 -0.823208   
1   -1.729703     -1.727516  -0.352454        0.254202  0.192843 -0.823208   
2   -1.728138     -1.725969  -0.352454       -0.674098  0.597087 -0.823208   
3   -1.726572     -1.724421  -0.352454        1.182503 -0.785073  1.039532   
4   -1.725007     -1.722874  -0.352454        0.254202 -1.222051  1.039532   

   Teenhome   Recency  MntWines  MntFruits  ...  AcceptedCmp3  AcceptedCmp4  \
0 -0.929307 -1.693697 -0.344435   1.955247  ...     -0.281979     -0.282912   
1 -0.929307 -1.693697  0.471034  -0.536764  ...     -0.281979     -0.282912   
2  0.908545 -1.693697 -0.507529  -0.385733  ...     -0.281979     -0.282912   
3  0.908545 -1.693697 -0.875231  -0.662623  ...     -0.281979     -0.282912   
4 -0.929307 -1.693697 -0.887093  -0.259874  ...      3.546363     -0.282912   

   AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain   Country       Age  \
0     -0.280107     -0.261851     -0.117229 -0.095498  0.694790 -0.092557   
1     -0.280107     -0.261851      8.530338 -0.095498 -1.600776  0.676833   
2     -0.280107     -0.261851     -0.117229 -0.095498  1.153903  0.933297   
3     -0.280107     -0.261851     -0.117229 -0.095498 -2.059889  0.163906   
4     -0.280107     -0.261851     -0.117229 -0.095498  0.694790 -1.716826   

   Total_Spent  customer_lifetime  
0     0.967836          -1.599987  
1    -0.049841          -1.599987  
2    -0.591052          -1.599987  
3    -0.989490          -1.420060  
4    -0.856677          -1.420060  

[5 rows x 29 columns]

In [43]:
df['Response'].value_counts()

0    1880
1     333
Name: Response, dtype: int64

In [46]:
y = df['Response']


### Splitting into train and test set

In [49]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=100)

In [50]:
y_train.value_counts()

0    1313
1     236
Name: Response, dtype: int64

### Logistic Regression

In [51]:
from sklearn.linear_model import LogisticRegression

logisticr = LogisticRegression().fit(X_train, y_train)

In [52]:
logisticr.score(X_test, y_test)

0.8810240963855421

In [53]:
train_predictions = logisticr.predict(X_train)
test_predictions = logisticr.predict(X_test)

Calculating and printing the accuracy score for both the training and the testing set of the marketing analysis

In [54]:
from sklearn.metrics import accuracy_score

print("Training of the marketing analysis:", accuracy_score(y_train, train_predictions))
print("Testing of the marketing analysis:", accuracy_score(y_test, test_predictions))

Training of the marketing analysis: 0.8967075532601678
Testing of the marketing analysis: 0.8810240963855421


In [55]:
from sklearn.metrics import roc_auc_score

In [56]:
roc_auc_score(y_test, test_predictions)

0.6953308241968036

The accuracy of my model in the test set without optimizing the hyperparameters is 69 %